Dependencies/Authentication/Sources/Authentication/AuthenticationManager.swift

```diff
@@ -0,0 +1,185 @@
+        AsyncStream(bufferingPolicy: .bufferingNewest(1)) { continuation in
+            loginEventHandler = { continuation.yield($0) }
+        }
+    }()
+    
+    func getAuthenticatedUser() throws -> Bool {
+        guard let auth = auth.currentUser else {
+            throw AuthErrorHandler.getAuthenticatedUserError
+        }
+        return true
+    }
+    
+    // MARK: Sign In with email
+    func createUser(email: String, password: String) async throws {
+        do {
+            try await auth.createUser(withEmail: email, password: password)
+        } catch {
+            throw AuthErrorHandler.signUpError
+        }
+    }
+    
+    func signInUser(email: String, password: String) async throws {
+        do {
+            let authDataResult = try await auth.signIn(withEmail: email, password: password).user
+            self.user = User(from: authDataResult)
+        } catch {
+            throw AuthErrorHandler.signInError
+        }
+    }
+    
+    func updatePassword(email: String, password: String, newPassword: String) async throws {
+        guard let user = auth.currentUser else {
+            throw AuthErrorHandler.updatePasswordError
+        }
+        try await user.updatePassword(to: password)
+    }
+    
+    func resetPassword(email: String) async throws {
+        do {
+            try await auth.sendPasswordReset(withEmail: email)
+        } catch {
+            throw AuthErrorHandler.resetPasswordError
+        }
+    }
+    
+    func deleteAccount() async throws {
+        guard let user = auth.currentUser else {
+            throw AuthErrorHandler.deleteUserError
+        }
+        try await user.delete()
+    }
+    
+    // MARK: PROVIDERS
+    func getProviders() throws -> [AuthenticationInterface.AuthProviderOption] {
+        guard let providerData = auth.currentUser?.providerData else {
+            throw AuthErrorHandler.getProvidersError
+        }
+        
+        var providers: [AuthProviderOption] = []
+        for provider in providerData {
+            if let option = AuthProviderOption(rawValue: provider.providerID) {
+                providers.append(option)
+                print(option)
+            } else {
+                assertionFailure("Provicer option not found: \(provider.providerID)")
+            }
+        }
+        return providers
+    }
+    
+    func signOut() throws {
+        do {
+            try auth.signOut()
+            self.user = nil
+        } catch {
+            throw AuthErrorHandler.signOutError
+        }
+    }
+    
+    // MARK: Sign In with SSO
+    func signIn(credential: AuthCredential) async throws -> AuthenticationInterface.User {
+        do {
+            let authDataResult = try await auth.signIn(with: credential)
+            self.user = User(from: authDataResult.user)
+            return User(from: authDataResult.user)
+        } catch {
+            throw AuthErrorHandler.signInWithCredentialError
+        }
+    }
```

Comment:
Nie wiem czy nazwanie tego błędem jest poprawnym określeniem 

Twoja funkcja zwraca -> `AuthenticationInterface.User`

Jeśli korzystasz z `private var user: User? { ... }` i twoje następne klasy bedą obserwować zmiany przy użyciu  `lazy var userUpdates: AsyncThrowingStream<User?, Error> = { ... }`

To nie ma sensu zwracać ->  `AuthenticationInterface.User` - chyba że faktycznie będziesz tworzyć paczkę User. 

Jeśli będziesz tworzyć paczkę User:
```suggestion
    func signIn(credential: AuthCredential) async throws -> AuthenticationInterface.User {
        do {
            let authDataResult = try await auth.signIn(with: credential)
            return User(from: authDataResult.user)
        } catch {
            throw AuthErrorHandler.signInWithCredentialError
        }
    }
```
Przykazujesz dane do paczki User 

Jeśli nie:  
```suggestion
    func signIn(credential: AuthCredential) async throws {
        do {
            let authDataResult = try await auth.signIn(with: credential)
            self.user = User(from: authDataResult.user)
        } catch {
            throw AuthErrorHandler.signInWithCredentialError
        }
    }
```
Twoja paczka "Home" obserwuje zmiany zachodzące w `authenticationManager` i korzystasz z AsyncStream

Reply:
wrócić